### ISSUE TITLE
[BUG] WPF XAML designer broken when child control supports activation

### ISSUE THREAD

**Issue by chuuddo** [reactiveui/ReactiveUI#1960]
**Describe the bug**
WPF XAML designer broken when child control implement IViewFor or IActivatable. 
Erorr message: 
```
Don't know how to detect when IActivationForViewFetcherBug.TabA is activated/deactivated, you may need to implement IActivationForViewFetcher
```
Link #1016 #1696  

**Steps To Reproduce**
Provide the steps to reproduce the behavior:
1. Clone https://github.com/chuuddo/IActivationForViewFetcherBug
2. Open MainWindow.xaml in XAML Designer
3. TabA and TabB broken, TabC not.

**Expected behavior**
XAML Designer should works.

**Screenshots**
![image](https://user-images.githubusercontent.com/526086/53623460-99df3500-3c16-11e9-97f2-d7a8e7c2113d.png)

**Environment(please complete the following information):**
 - Win 10 1809
 - VS 2019 Preview 4
 - ReactiveUI.WPF 9.11.1

**Comment by glennawatson** [comment]
Don't believe this is IViewFor<T> it'd be the fact you're using the generic base class `ReactiveUserControl<TViewModel>` and that the XAML designer doesn't support generics properly. (may be worth lodging a bug about that directly with MSFT)

https://reactiveui.net/api/reactiveui/reactiveusercontrol_1/#Remarks for some work around details.

**Comment by chuuddo** [comment]
@glennawatson TabB not use generic interfaces, but still broken due to lack of IActivationForViewFetcher

**Comment by glennawatson** [comment]
@chuuddo do these issues happen with VS2017 (for the non-generic version of the issues)

This may be a Splat issue with the test mode/design mode checker is all otherwise and I may end up transferring the issue there.

The generic design mode bug unfortunately is outside our control.

**Comment by chuuddo** [comment]
@glennawatson dont have installed VS2017 and cant test, maybe some else can.

**Comment by glennawatson** [comment]
no problem, I will take a look next week into the issue (have another one that is a bit higher priority at the moment).

**Comment by MarkusKgit** [comment]
I think the WPF designer is hopelessly broken. One can only hope they update it as part of the WPF net core 3.0 initiative. 
For me it is initially not working, but when I rebuild the project with Tab C as the active Tab, then I can switch to Tab A just fine. If I then rebuild again with Tab A active then sometimes it fails again. I can't seem to reproduce constant results though... I have tried with VS 2017 (latest) and VS 2019 RC1

**Comment by chuuddo** [comment]
Just try workaround from 
https://reactiveui.net/api/reactiveui/reactiveusercontrol_1/#Remarks
but still broken for my environment.
Added TabD to mensioned repository.

**Comment by CodingNinjaOctocat** [comment]
By the way, what type is TabA/B/C? a Page? I can't add a Page in TabItem.

**Comment by chuuddo** [comment]
#2180 fixes this issue on VS 16.3.0.

**Comment by glennawatson** [comment]
Closing due to confirmed fix.
